src/ApiCompat/Microsoft.DotNet.ApiCompatibility/ApiComparer.cs

```diff
@@ -38,7 +37,7 @@ public class ApiComparer : IApiComparer
 
             DifferenceVisitor visitor = new();
             visitor.Visit(mapper);
-            return visitor.DiagnosticCollections.First();
+            return visitor.DiagnosticCollections[0];
```

Comment:
I know this is not new from your work, but do we have to check if this is null? or if the size of the DiagnosticCollection has at least 1 elemnt?

Reply:
The size of the `DiagnosticCollection` is governed by the `rightCount` ctor parameterof the `DifferenceVisitor` type and that value defaults to 1: https://github.com/dotnet/sdk/blob/1c6bef9ef27528bb2dbf48dd890c8668641182a1/src/ApiCompat/Microsoft.DotNet.ApiCompatibility/DifferenceVisitor.cs#L28.

Also, the individual values in the collection are already initialized to an empty `HashSet<CompatDifference>`, so this could never be null.